### PR TITLE
Add additional fields for seller and buyer in wishlist

### PR DESCRIPTION
**Pull request by sabatohura**
#### What does this PR do?

- Provide additional product information to the buyer's wishlist.

#### Description of Task to be completed?

- Added Product Name, thumbnail image and productPrice

#### How should this be manually tested?

- in existing API swagger, **api/wishes**

### CODE REVIEW

**Review by kayigmb — APPROVED** [review]

**Review by emmanueltct — APPROVED** [review]

File: src/app.ts [hunk 1]
```diff
@@ -66,12 +66,19 @@ app.use("/api/users", chatRouters);
 app.use("/api/users/", authRoutes);
 app.get("/", authenticate, Home);
 app.use("/api/users", userRoutes);
-app.use("/api/roles", authenticate, checkRole(["admin"]), roleRoutes);
+app.use(
+  "/api/roles",
+  authenticate,
+  checkRole(["admin"]),
+  authenticate,
+  checkRole(["admin"]),
```
Comment (Eli250) [inline src/app.ts]: Why are these middlewares duplicated ?

File: src/app.ts [hunk 2]
```diff
@@ -66,12 +66,19 @@ app.use("/api/users", chatRouters);
...
+  roleRoutes
+);
 app.use("/api/users/", authRoutes);
 app.use("/api/search/", searchRoutes);
 app.use("/api/carts/", authenticate, checkRole(["buyer"]), cartRouter);
 
 app.use("/api/products", productRoutes);
 app.use("/api/wishes", authenticate, productWishRoutes);
-app.use("/api/categories", authenticate, productCategoryRoutes);
+app.use("/api/categories", authenticate, authenticate, productCategoryRoutes);
```
Comment (Eli250) [inline src/app.ts]: Do not duplicate middlewares

File: src/controllers/wish.controller.ts [hunk 3]
```diff
@@ -1,66 +1,93 @@
...
-      res.status(200).send({ message: "product wish list is empty" });
+    const role = await Role.findByPk(userRole);
+    let wishesData: any[] = [];
+
+    if (role?.dataValues.name == "buyer") {
+      const wishes: any = await wishServices.getUserWishes(userId);
+      wishesData = await Promise.all(
+        wishes.map(async (wish: any) => {
+          const { productId, userId, updatedAt, ...wishData } = wish.dataValues;
+          const product = await Product.findByPk(productId, {
```
Comment (Eli250) [inline src/controllers/wish.controller.ts]: Do not call database inside the loop. Instead, generate an array of product ids from wishes object and use that array in `Product.findAll` method, this way you will get information of the products matching the ids in the array without looping through database.

**Review by Eli250 — CHANGES_REQUESTED** [review]
Resolve feedback.

File: src/app.ts (hunk 2 above)
Comment (sabatohura) [reply src/app.ts]: Resolved

File: src/controllers/wish.controller.ts (hunk 3 above)
Comment (sabatohura) [reply src/controllers/wish.controller.ts]: Resolved

File: src/app.ts (hunk 1 above)
Comment (sabatohura) [reply src/app.ts]: this is also resolved

**Review by Eli250 — APPROVED** [review]

**Pull request merged**